docs/docs/development/events.md

```diff
@@ -0,0 +1,178 @@
+---
+title: ZMK Events
+sidebar_label: ZMK Events
+---
+
+ZMK makes use of events to decouple individual components such as behaviors and peripherals from the core functionality. For this purpose, ZMK has implemented its own event manager. This page is a (brief) overview of the functionality and methods exposed by the event manager. To see what events exist and what data they contain, it is best to view the corresponding [event header files](https://github.com/zmkfirmware/zmk/tree/main/app/include/zmk/events) directly.
+
+Including `zmk/event_manager.h` is required for any interaction with the event system.
+
+## Generic Events
```

Comment:
I keep going back and forth on whether this section is necessary for users, since it goes into `ZMK_EVENT_DECLARE`/`ZMK_EVENT_IMPL` implementation details and is confusing to me. I guess it is necessary for wherever we mention `zmk_event_t`, "header", "data" etc. below.